src/21-pubsub/index.md

```diff
@@ -0,0 +1,134 @@
+---
+# this is the final blog post's id (used in the directory)
+# this id is part of the url, and should only contain:
+#   letters, numbers, dashes.
+id: 25-pubsub
+
+breadcrumbs:
+  - {name: "Take a look at pubsub on IPFS", link: "./" }
+
+# the date here should be set to the final publication date,
+# on the day it is published.
+date: 2017-05-17
+
+# this is the Title
+title: Take a look at pubsub on IPFS
+
+# this is the name of the main author(s)
+author: Jeromy Johnson
+
+# technical details required for the software, don't change these.
+baseurl: ..
+template: tmpl/layouts/post.html
+collection: posts
+---
+
+We recently merged a simple, experimental pubsub implementation into IPFS. This
+implementation is just a beginning. It is far from the performance and security goals
+we will achieve in our long-term target. However, even this early implementation opens
+the doors to several useful and interesting new applications.
+
+In this post, I will point out some applications of this technology, show how to
+get started started using `ipfs pubsub`, and discuss upcoming improvements.
+
+## Why pubsub?
+[Publish-Subscribe](https://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern),
+called 'pubsub' for short, is a pattern often used to handle events in
+large-scale networks. 'Publishers' send messages classified by topic or content and
+'subscribers' receive only the messages they are interested in, all without direct
+connections between publishers and subscribers. This approach offers much greater
+network scalability and flexibility. 
+
+Pubsub makes IPFS fast for large-scale networks such as datacenters, local area
```

Comment:
Some applications include as collaborative document editing, "dynamic" website content, chat applications, multiplayer games, continuously evolving datasets, and webservice workers passing around messages. It gives us ways to make IPFS fast for large-scale networks such as datacenters, local area networks, and large p2p applications. In the near future, IPNS records will be pushed over pubsub, allowing lightning fast updates of peers' IPNS entries. Peers could use pubsub to track the head of a merkle-linked global log.